packages/catalog-search/src/SearchBox.jsx

```diff
@@ -54,40 +59,61 @@ export const SearchBoxBase = ({
     }
   };
 
+  const handleSuggestionSubmit = (query, hit) => {
+    // If an override is provided, call it with the hit
+    if (suggestionSubmitOverride !== undefined) {
```

Comment:
[curious] Could `suggestionSubmitOverride` ever be `null` here? Or does passing `null` convert to `undefined` due to the default prop? 🤔

Reply:
hmmm let me check!

Reply:
Hmm so, simply not providing the prop will allow things to work as expected because the default value forces it to undefined. However, you can break this by manually passing `null` in through the `suggestionSubmitOverride` prop. That should be ok though, right? Because why would someone force it to null when they could just not provide the prop?

Reply:
since it's a function, I think the safest bet is to do this?

```if (typeof suggestionSubmitOverride === 'function')```

this way it will be even safer to null ,undefined or any other non functoin value